Lib/test/test_types.py

```diff
@@ -672,6 +672,25 @@ def test_traceback_and_frame_types(self):
     def test_capsule_type(self):
         self.assertIsInstance(_datetime.datetime_CAPI, types.CapsuleType)
 
+    def test_call_unbound_crash(self):
+        # GH-131998: The specialized instruction would get tricked into dereferencing
+        # a bound "self" that didn't exist if subsequently called unbound.
+        code = """if True:
+        # The optimizer is finicky, and the easiest way (that I know of)
+        # to get it to reproduce in CI is by importing a Python module at runtime.
+        import glob
+
+        def call(part):
```

Comment:
```suggestion
        def call(part):
            [] + ([] + [])
```

To trigger the failure you'll need to leave an empty list on the stack. This seems to work.